### ISSUE TITLE
Audio wave normalization has unexpected behaviour when using the `normalize` Wavesurfer option while also using the `ZoomPlugin`

### ISSUE THREAD

**Issue by filipkoukal** [katspaugh/wavesurfer.js#3554]
## Bug description
Audio wave normalization doesn't work as expected when using the `normalize` Wavesurfer option while also using the `ZoomPlugin`. 
How to reproduce:
1. Zoom in on an area 
2. Move right or left along the audio wave to view a before unseen part of audio wave

## Environment
 - Browser: Chrome

## Minimal code snippet
```
import WaveSurfer from 'wavesurfer.js'
import ZoomPlugin from 'wavesurfer.js/dist/plugins/zoom';


const wavesurfer = WaveSurfer.create({
    container: document.body,
    waveColor: 'rgb(200, 0, 200)',
    progressColor: 'rgb(100, 0, 100)',
    url: '/examples/audio/demo.wav',
    normalize: true,
})

wavesurfer.registerPlugin(ZoomPlugin.create())

wavesurfer.on('interaction', () => {
  wavesurfer.play()
})
```

## Expected result
The audio wave is stable even when moving left or right along the wave when using the `normalize` option.

## Obtained result
When moving left or right along the zoomed audio wave, previously unseen parts aren't normalized correctly.

## Screenshots
<img width="1312" alt="1" src="https://github.com/katspaugh/wavesurfer.js/assets/60878671/793a102f-3264-4743-a100-9b4cfa9a4956">
<img width="1312" alt="2" src="https://github.com/katspaugh/wavesurfer.js/assets/60878671/364d65e0-ed23-4163-8d6e-8cab29ea59c2">
<img width="1318" alt="3" src="https://github.com/katspaugh/wavesurfer.js/assets/60878671/98a3242f-5c78-4ddb-8d38-3a7af4216523">

**Comment by katspaugh** [comment]
Good catch. The reason is the normalization is performed on render, for each visible chunk individually. It should instead normalize by max value across the entire array of peaks.

**Comment by Schwankenson** [comment]
Hi, any news or monkeypatch here? I think the normalization feature is not usable like that. Never seen an audio editor which normalizes in current zoomed audio. It`s very unexpected zooming into a quieter part and then it is shown width 100% height

**Comment by katspaugh** [comment]
I'm currently busy but anyone is welcome to contribute. The code to blame is [here](https://github.com/katspaugh/wavesurfer.js/blob/main/src/renderer.ts#L466-L469).

This should be moved to wavesurfer.ts after the audio is decoded and sent to the renderer.

**Comment by yottanami** [comment]
This looks fixed on `main` already, as part of the v8.0.0 pre-release refactor (renderer.ts's `renderChannel` now computes `calculateGlobalPeak(channelData)` once and threads it through as `maxPeak` into `calculateVerticalScale`, instead of recomputing per visible canvas slice), see 8ca8693 / PR #4354. Looks like this issue can be closed, or let me know if there's a remaining case it doesn't cover.

**Comment by katspaugh** [comment]
Thanks, @yottanami! Closing then.
